### PR TITLE
feat(metrics): add Character Error Rate (CER) metric

### PR DESCRIPTION
**Pull request by Manimaran-tech**
## Description

This PR adds a new NLP metric, `CharacterErrorRate`, to Ignite.

Character Error Rate (CER) measures the normalized edit distance between predicted and reference strings at the character level.

### What’s included

- Adds `CharacterErrorRate` metric implementation
- Uses Levenshtein edit distance to compute character-level errors
- Supports batch-wise updates and distributed usage
- Follows Ignite metric API conventions used by existing NLP metrics

### Why

CER is a standard evaluation metric for ASR/OCR and other sequence-to-text tasks where character-level accuracy is important.

Fixes #{issue_number}

## Checklist

- [x] New tests are added (if a new feature is added)
- [x] New docstrings are added and in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/nlp/character_error_rate.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        refs = 0.0
+        for p, r in zip(y_pred, y):
+            p_chars = list(p)
+            r_chars = list(r)
+
+            errors += _edit_distance(r_chars, p_chars)
+            refs += len(r_chars)
+
+        self._num_errors += torch.tensor(errors, device=self._device)
+        self._num_refs += torch.tensor(refs, device=self._device)
```
Comment (aaishwarymishra) [inline ignite/metrics/nlp/character_error_rate.py:116]: I think we should add a check for p and r is string or not .

**Comment by aniketpandey05** [comment]
@Manimaran-tech, could you link the issue number for this PR , also the Empty reference strings aren't handled in update and Distributed integration tests are missing as the other metric bleu and rogue have this

**Comment by rogueslasher** [comment]
@vfdev-5 should i update the pr or this one is good to be merged?
